### PR TITLE
Return projectionSchema status for read operations 

### LINKED ISSUE #1584: Return the schema of the projection for read commands
Per the spec, return the schema in the status using the same format as we do for inserts. 

Example from insert response below: 

```
{
    "status": {
        "insertedIds": [
            [
                "aaron.morton@datastax.com"
            ],
            [
                "alice@datastax.com"
            ]
        ],
        "primaryKeySchema": {
            "email": {
                "type": "text"
            }
        }
    }
}
```

The pattern started with the insert was to have a status key specific for reads and insert, so call it  ?? `projectionSchema' ??

### PR DESCRIPTION
**Pull request by amorton**
Fixes #1584

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiDataTypeDefs.java [hunk 1]
```diff
@@ -1,88 +1,303 @@
...
+  // below.
+  static final List<PrimitiveApiDataTypeDef> PRIMITIVE_TYPES =
       List.of(
           ASCII, BIGINT, BOOLEAN, BINARY, DATE, DECIMAL, DOUBLE, DURATION, FLOAT, INT, SMALLINT,
           TEXT, TIME, TIMESTAMP, TINYINT, VARINT, INET, UUID, TIMEUUID);
 
-  public static final Map<DataType, ApiDataTypeDef> PRIMITIVE_TYPES_BY_CQL_TYPE =
-      PRIMITIVE_TYPES.stream()
-          .collect(Collectors.toMap(ApiDataTypeDef::getCqlType, Function.identity()));
+  //  private static final Map<ApiDataType, PrimitiveApiDataTypeDef> CQL_TYPES_BY_PRIMITIVE_TYPE =
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiDataTypeDefs.java:73]: Remove the commented code

File: src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiListType.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import com.datastax.oss.driver.api.core.type.ListType;
+import com.datastax.oss.driver.internal.core.type.PrimitiveType;
+import io.stargate.sgv2.jsonapi.api.model.command.table.definition.datatype.ListColumnDesc;
+import io.stargate.sgv2.jsonapi.exception.checked.UnsupportedCqlType;
+import io.stargate.sgv2.jsonapi.exception.checked.UnsupportedUserType;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.VectorizeDefinition;
+import io.stargate.sgv2.jsonapi.service.resolver.VectorizeConfigValidator;
+import java.util.Objects;
+
+public class ApiListType extends CollectionApiDataType {
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiListType.java:13]: Java doc

File: src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiMapType.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import io.stargate.sgv2.jsonapi.api.model.command.table.definition.datatype.MapColumnDesc;
+import io.stargate.sgv2.jsonapi.exception.checked.UnsupportedCqlType;
+import io.stargate.sgv2.jsonapi.exception.checked.UnsupportedUserType;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.VectorizeDefinition;
+import io.stargate.sgv2.jsonapi.service.resolver.VectorizeConfigValidator;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApiMapType extends CollectionApiDataType {
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiMapType.java:15]: Java Doc

**Review by maheshrajamani — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/ColumnDescDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    // if the nodes are missing, the jackson MissingNode will be returned and has "" and 0 for
+    // defaults.
+    // but to get a decent error message get the dimension as a string
+    // arguable that a non integer is a JSON mapping error, but we will handle it as an unsupported
+    // dimension value
+    var keyTypeName = descNode.path(TableDescConstants.ColumnDesc.KEY_TYPE).asText();
+    var valueTypeName = descNode.path(TableDescConstants.ColumnDesc.VALUE_TYPE).asText();
+    var dimensionString = descNode.path(TableDescConstants.ColumnDesc.DIMENSION).asText();
```
[the remark this answers is not in the document]
Comment (amorton) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/ColumnDescDeserializer.java:95]: comments above saying reading as a string to get a better error message, you have a test that passed "aaa" as the dimension. This gives a consistent invalid dimension value for numbers out of range and non numbers

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/serializer/ColumnDescSerializer.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Override
+  public void serialize(
+      ColumnDesc columnDesc, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+      throws IOException {
+
+    jsonGenerator.writeStartObject();
+    jsonGenerator.writeStringField(TableDescConstants.ColumnDesc.TYPE, columnDesc.getApiName());
+
+    if (columnDesc instanceof MapColumnDesc mt) {
```
[the remark this answers is not in the document]
Comment (amorton) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/serializer/ColumnDescSerializer.java:24]: cool, will do

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/PrimaryKeyDesc.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * <b>NOTE:</b> use the `from()` functions to construct when you have the CQL Identifiers so there
+ * is a guarantee that the keys are properly formatted.
+ *
+ * @param keys
+ * @param orderingKeyDescs
+ */
+@JsonDeserialize(using = PrimaryKeyDescDeserializer.class)
+// TODO, hide table feature detail before it goes public,
```
[the remark this answers is not in the document]
Comment (amorton) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/PrimaryKeyDesc.java:30]: will do

File: src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiDataTypeDefs.java (hunk 1 above)
Comment (amorton) [reply src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiDataTypeDefs.java:73]: will do

**Pull request merged**